protocol/x/clob/keeper/liquidations.go

```diff
@@ -161,20 +161,45 @@ func (k Keeper) MaybeGetLiquidationOrder(
 	)
 
 	// The subaccount is liquidatable. Get the perpetual position and position size to liquidate.
-	perpetualId, baseQuantumsBig, err := k.GetPerpetualPositionToLiquidate(ctx, subaccountId)
+	perpetualId, err := k.GetPerpetualPositionToLiquidate(ctx, subaccountId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.GetLiquidationOrderForPerpetual(
+		ctx,
+		subaccountId,
+		perpetualId,
+	)
+}
+
+// GetLiquidationOrderForPerpetual returns a liquidation order for a subaccount
+// given a perpetual ID and position size delta.
+func (k Keeper) GetLiquidationOrderForPerpetual(
+	ctx sdk.Context,
+	subaccountId satypes.SubaccountId,
+	perpetualId uint32,
+) (
+	liquidationOrder *types.LiquidationOrder,
+	err error,
+) {
+	deltaQuantums, err := k.GetLiquidatablePositionSizeDelta(
+		ctx,
+		subaccountId,
+		perpetualId,
+	)
 	if err != nil {
 		return nil, err
 	}
 
 	// Get the fillable price of the liquidation order in subticks.
-	deltaQuantumsBig := baseQuantumsBig.Neg(baseQuantumsBig)
```

Comment:
not immediately clear to me why this was removed

Reply:
ah this was moved to `GetLiquidatablePositionSizeDelta`.

GetPerpetualPositionToLiquidate used to return position size in base quantums. and delta was calculated outside of the function.